t/05-scheduler-dependencies.t

```diff
@@ -1350,6 +1388,58 @@ subtest 'WORKER_CLASS validated when creating directly chained dependencies' =>
     );
 };
 
+subtest 'skip passed children' => sub {
+    # create a cluster
+    #                -> child-1-passed
+    #               /
+    # parent-passed --> child-2-passed --> child-2-child-1-failed
+    #               \
+    #                -> child-3-failed
+    my $parent          = _job_create({%settings, TEST => 'parent-passed'});
+    my $child_1         = _job_create({%settings, TEST => 'child-1-passed'}, undef, undef, [$parent->id]);
+    my $child_2         = _job_create({%settings, TEST => 'child-2-passed'}, undef, undef, [$parent->id]);
+    my $child_2_child_1 = _job_create({%settings, TEST => 'child-2-child-1-failed'}, undef, undef, [$child_2->id]);
+    my $child_3         = _job_create({%settings, TEST => 'child-3-failed'}, undef, undef, [$parent->id]);
+    my @all_jobs = ($parent, $child_1, $child_2, $child_2_child_1, $child_3);
+    my $log_jobs = sub { note $_->TEST . ': id=' . $_->id . ', clone_id=' . ($_->clone_id // 'none') for @all_jobs };
+    $_->update({state => DONE, result => PASSED}) for ($parent, $child_1, $child_2);
+    $_->update({state => DONE, result => FAILED}) for ($child_2_child_1, $child_3);
+    $_->discard_changes for @all_jobs;
+
+    # duplicate parent
+    $parent->auto_duplicate({skip_passed_children => 1});
```

Comment:
Maybe add `isnt(undef,...` here to check that we have a clone?

Reply:
That's asserted in the next subtest anyways. As mentioned in the previous comment the error handling should be generally improved (to throw an exception in any error case).